Fixes a problem of ambiguous table names when using only_deleted method and joining tables that have a scope on `with_deleted`

thromera opened the pull request:
Fixes #26 and #27 (that are still present). 

This bug would happen when overriding sentinel with `active` field

```
# paranoid.rb
module Paranoid
  extend ActiveSupport::Concern

  # See index section here: https://github.com/radar/paranoia

  included do
    acts_as_paranoid column: :active, sentinel_value: true
    include Overrides
  end

  # If not defined in a separate module, Paranoia will be prepended before Paranoid, causing these methods to be later
  # in the call chain
  module Overrides
    def paranoia_restore_attributes
      {
        deleted_at: nil,
        active: true
      }
    end

    def paranoia_destroy_attributes
      {
        deleted_at: current_time_from_proper_timezone,
        active: nil
      }
    end
  end
end
```

```
#post.rb
class Post < ActiveRecord::Base
    include Paranoid

    belongs_to :author, -> {with_deleted}
    belongs_to :event, -> {with_deleted}

    validates_presence_of :author, :event
end
```

```
#author.rb
class Author < ActiveRecord::Base
    include Paranoid

    has_many :posts, -> {with_deleted}, dependent: :destroy
end
```

```
#event.rb
class Event < ActiveRecord::Base
    include Paranoid

    has_many :posts, -> {with_deleted}
end
```

```
Post.only_deleted.joins(:event)
2.2.3 :001 > Post.only_deleted.joins(:event)
  Post Load (10.1ms)  SELECT "posts".* FROM "posts" INNER JOIN "events" ON "events"."id" = "posts"."event_id" AND "events"."active" = ? WHERE ("active" IS NULL OR "active" != 't')  [["active", "t"]]
ActiveRecord::StatementInvalid: SQLite3::SQLException: ambiguous column name: active: SELECT "posts".* FROM "posts" INNER JOIN "events" ON "events"."id" = "posts"."event_id" AND "events"."active" = ? WHERE ("active" IS NULL OR "active" != 't')
```

thromera committed:
Fixes a problem of ambiguous table names when using only_deleted method and joining tables that have a scope on `with_deleted`

Update homepage in gemspec

The homepage is supposed to be where you can find the code. It is displayed on rubygems.org page. Currently it will redirect you back to the same page. Should point at this github repo.

Update README to use proper version for Rails 5

Version 2.2.0

Ignore failures from all jruby's on travis

Add explicit language about dependent: :destroy

Update CHANGELOG.md

update ruby and rails versions

Use ActiveSupport.on_load to correctly re-open ActiveRecord::Base. https://github.com/rubysherpas/paranoia/issues/335

Touch record on paranoia-destroy. Fixes #296

Touch record on destroy by leveraging the paranoia_destroy_attributes.
Applied the same to the restore-method as this eliminates the extra query.

schneems committed:
Update homepage in gemspec

The homepage is supposed to be where you can find the code. It is displayed on rubygems.org page. Currently it will redirect you back to the same page. Should point at this github repo.

noahmatisoff committed:
Update README to use proper version for Rails 5

jhawthorn committed:
Version 2.2.0

jhawthorn committed:
Ignore failures from all jruby's on travis

supremebeing7 committed:
Add explicit language about dependent: :destroy

BenMorganIO committed:
update ruby and rails versions

dlackty committed:
Update CHANGELOG.md

iaankrynauw committed:
Use ActiveSupport.on_load to correctly re-open ActiveRecord::Base. https://github.com/rubysherpas/paranoia/issues/335

BenMorganIO commented on test/paranoia_test.rb:
```diff
@@ -5,6 +5,7 @@
 
 test_framework = defined?(MiniTest::Test) ? MiniTest::Test : MiniTest::Unit::TestCase
 
+
```
?

BenMorganIO commented on test/paranoia_test.rb:
```diff
@@ -1105,6 +1122,47 @@ def paranoia_destroy_attributes
+end
+
+
+
```
?

BenMorganIO commented:
The commit for this doesn't lead to a Github profile and isn't owned by anyone. I'm a bit skeptical security wise.

thromera commented:
I'm currently in vacation, I will fix that next week, my git config was probably wrong at that time.

thromera commented:
It should be fixed now. (I've also fixed the useless new lines.)

BenMorganIO commented:
Cool. I'd love to get this in, but it looks like you need to rebase the latest changes from the `core` branch.

Commit:
Touch record on paranoia-destroy. Fixes #296

Touch record on destroy by leveraging the paranoia_destroy_attributes.
Applied the same to the restore-method as this eliminates the extra query.

thromera commented:
I'm closing this PR since I've messed up with my rebase. Sorry about that. 
Good news is https://github.com/rubysherpas/paranoia/pull/379 should be OK to merge.
